be/src/storage/table_reader.cpp

```diff
@@ -14,22 +14,121 @@
 
 #include "storage/table_reader.h"
 
+#include "column/datum_convert.h"
+#include "column/datum_tuple.h"
+#include "storage/chunk_helper.h"
+#include "storage/projection_iterator.h"
+#include "storage/storage_engine.h"
+#include "storage/tablet_manager.h"
+#include "storage/tablet_reader.h"
+#include "storage/union_iterator.h"
+
 namespace starrocks {
 
-TableReader::TableReader(const TableReaderParams& params) : _params(params) {}
+TableReader::TableReader(const TableReaderParams& params) : _params(params) {
+    for (auto& partition : _params.partition_param.partitions) {
+        for (auto& index : partition.indexes) {
+            for (int64_t tablet_id : index.tablets) {
+                TabletSharedPtr tablet = StorageEngine::instance()->tablet_manager()->get_tablet(tablet_id, true);
+                if (tablet != nullptr) {
+                    _local_tablets.push_back(tablet);
+                }
```

Comment:
What if tablet is nullptr? throw exception or add warning?

Reply:
If there is no local tablets, multi_get will return not found, and scan will return empty iterator